laws/src/main/scala/cats/laws/discipline/MiniInt.scala

```diff
@@ -0,0 +1,36 @@
+package cats
+package laws
+package discipline
+
+import cats.instances.int._
+
+/**
+ * Similar to `Int`, but with a much smaller domain. The exact range of [[MiniInt]] may be tuned
+ * from time to time, so consumers of this type should avoid depending on its exact range.
+ */
+final class MiniInt private (val asInt: Int) extends AnyVal with Serializable
```

Comment:
Since this `MiniInt` will be used in type parameters a lot, maybe we should give https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/Newtype.scala a try to see if that makes an impact in test? Of course that could be in a separate PR.

Reply:
I would imagine that this overhead would be pretty small compared to other things that tests are doing, but it's certainly something that we could explore.